stormpath/express-stormpath
# Upgrade JSON error responses

This PR upgrades JSON error responses according to the [framework spec](https://github.com/stormpath/stormpath-framework-spec/blob/master/error-handling.md#error-handling).

It doesn't handle any Node errors that might occur due to e.g. network errors and bugs. These should be caught and sent as status `500 Internal Server Error`. This will have to be addresses in another PR.

Fixes #325.

### Notes

It can be a bit hard to actually verify all errors, so try a few and then put in some extra time into reviewing the code.

### How to verify

1. Checkout this branch
2. Use Postman or another HTTP tool and set the Accept header to `application/json`
3. Call the API and try to provoke different errors (example on `/login` and `/register`)
4. Verify that the errors have the same format as in the spec:

  ```javascript
{
    "status": 400,
    "message": "First Name required."
}
```
